cmd/hostpaths/hostpaths.go

```diff
@@ -420,6 +433,32 @@ func mapHostPaths(ctx context.Context, pManager, vManager manager.Manager) {
 	}, time.Second*5)
 }
 
+func filterManagedPods(ctx context.Context, options *context2.VirtualClusterOptions, podList *corev1.PodList, pManager manager.Manager) error {
+	if options.MultiNamespaceMode {
+		// Limit Pods
+		nsList := &corev1.NamespaceList{}
+		err := pManager.GetClient().List(ctx, nsList, &client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set{
+				namespaces.VclusterNamespaceAnnotation: options.TargetNamespace,
+			}),
+		})
+		if err != nil {
+			return errors.Wrap(err, "unable to list namespaces")
+		}
+
+		managedNamespaces := make([]string, 0, len(nsList.Items))
+		for _, ns := range nsList.Items {
+			managedNamespaces = append(managedNamespaces, ns.Name)
+		}
+
+		podList.Items = slices.DeleteFunc[[]corev1.Pod, corev1.Pod](podList.Items, func(pod corev1.Pod) bool {
```

Comment:
Isn't one be able to write this as the following?

```suggestion
		podList.Items = slices.DeleteFunc(podList.Items, func(pod corev1.Pod) bool {
```

Reply:
Yup, it works Don't know why I was verbose. But this is not needed anymore.